packages/datagrid/src/components/datagrid.component.js

```diff
@@ -0,0 +1,246 @@
+import theme from './datagrid.scss';
+
+const FOCUSED_COLUMN_CLASS_NAME = 'column-focus';
+const AG_GRID_CUSTOM_HEADER_KEY = 'headerComponent';
+const AG_GRID_CUSTOM_CELL_KEY = 'cellRenderer';
+const AG_GRID_DEFAULT_ROW_SELECTION = 'single';
+const HEADER_HEIGHT = 69;
+const ROW_HEIGHT = 39;
+
+export function injectedHeaderRenderer(getComponent, headerRenderer, onFocusedColumn) {
+	const Component = Inject.get(getComponent, headerRenderer);
+
+	return props => <Component {...props} onFocusedColumn={onFocusedColumn} />;
+}
+
+export function injectedCellRenderer(getComponent, cellRenderer, avroRenderer) {
+	const Component = Inject.get(getComponent, cellRenderer);
+
+	return props => <Component {...props} avroRenderer={avroRenderer} getComponent={getComponent} />;
+}
+
+function getAvroRenderer(avroRenderer) {
+	return {
+		booleanCellRenderer: 'DefaultBooleanCellRenderer',
+		dateCellRenderer: 'DefaultDateCellRenderer',
+		intCellRenderer: 'DefaultIntCellRenderer',
+		stringCellRenderer: 'DefaultStringCellRenderer',
+		...avroRenderer,
+	};
+}
+
+export default class DataGrid extends React.Component {
+	static defaultProps = {
+		cellRenderer: 'DefaultCellRenderer',
+		getPinnedColumnDefsFn: sampleSerializer.getPinnedColumnDefsFromSample,
+		getColumnDefsFn: sampleSerializer.getColumnDefsFromSample,
+		getRowDataFn: sampleSerializer.getRowDataFromSample,
+		getValueGetterFn: sampleSerializer.valueGetterFromRowData,
+		headerHeight: HEADER_HEIGHT,
+		headerRenderer: 'DefaultHeaderRenderer',
+		pinHeaderRenderer: 'DefaultPinHeaderRenderer',
+		rowHeight: ROW_HEIGHT,
+		rowSelection: AG_GRID_DEFAULT_ROW_SELECTION,
+	};
+
+	static propTypes = DATAGRID_PROPTYPES;
+
+	constructor(props) {
+		super(props);
+
+		this.handleKeyboard = this.handleKeyboard.bind(this);
+		this.onFocusedColumn = this.onFocusedColumn.bind(this);
+		this.onFocusedCell = this.onFocusedCell.bind(this);
+		this.onGridReady = this.onGridReady.bind(this);
+		this.setGridElement = this.setGridElement.bind(this);
+		this.setCurrentFocusedColumn = this.setCurrentFocusedColumn.bind(this);
+		this.updateStyleFocusColumn = this.updateStyleFocusColumn.bind(this);
+	}
+
+	onGridReady({ api }) {
+		this.gridAPI = api;
+	}
+
+	onFocusedCell({ column, ...rest }) {
+		if (!column) {
+			return;
+		}
+
+		if (column.colId !== this.currentColId || column.pinned) {
+			this.removeFocusColumn();
+		}
+
+		this.setCurrentFocusedColumn(column.colId);
+
+		if (column.pinned) {
+			return;
+		}
+
+		this.updateStyleFocusColumn();
+
+		if (this.props.onFocusedCell) {
+			this.props.onFocusedCell({
+				column,
+				...rest,
+			});
+		}
+	}
+
+	onFocusedColumn(colId) {
+		let selectedRowIndex = 0;
+		if (this.gridAPI.getFocusedCell()) {
+			selectedRowIndex = this.gridAPI.getFocusedCell().rowIndex;
+		}
+
+		if (colId !== this.currentColId) {
+			this.removeFocusColumn();
+		}
+
+		this.setCurrentFocusedColumn(colId);
+		this.gridAPI.setFocusedCell(selectedRowIndex, colId);
```

Comment:
Why do we keep the row to select a cell ? Is it a UX requirement ?
Shouldn't we just select the column on column selection and remove the selected cell ? (cf TDP)

Reply:
For the moment, I developed this data grid with requirement for TDC and the issue met by TDS. There is some adjustment for all products. 
On TDS, when we change the column, the selected rows is kept or the first row is selected.

Reply:
Please make the change, saw with PM/UX